components/brave_rewards/core/contribution/contribution_unblinded.cc

```diff
@@ -233,47 +235,45 @@ void Unblinded::PrepareTokens(
     token_id_list.push_back(base::NumberToString(item.id));
   }
 
-  auto reserved_callback = std::bind(
-      &Unblinded::OnMarkUnblindedTokensAsReserved, this, _1,
-      std::move(token_list),
-      std::make_shared<mojom::ContributionInfoPtr>(contribution->Clone()),
-      types, callback);
-
   engine_->database()->MarkUnblindedTokensAsReserved(
-      token_id_list, contribution_id, reserved_callback);
+      token_id_list, contribution_id,
+      base::BindOnce(&Unblinded::OnMarkUnblindedTokensAsReserved,
+                     weak_factory_.GetWeakPtr(), std::move(token_list),
+                     std::move(contribution), std::move(types),
+                     std::move(callback)));
 }
 
 void Unblinded::OnMarkUnblindedTokensAsReserved(
-    mojom::Result result,
-    const std::vector<mojom::UnblindedToken>& unblinded_tokens,
-    std::shared_ptr<mojom::ContributionInfoPtr> shared_contribution,
-    const std::vector<mojom::CredsBatchType>& types,
-    LegacyResultCallback callback) {
+    std::vector<mojom::UnblindedToken> unblinded_tokens,
+    mojom::ContributionInfoPtr contribution,
+    std::vector<mojom::CredsBatchType> types,
+    ResultCallback callback,
+    mojom::Result result) {
   if (result != mojom::Result::OK) {
     engine_->LogError(FROM_HERE) << "Failed to reserve unblinded tokens";
-    callback(mojom::Result::FAILED);
+    std::move(callback).Run(mojom::Result::FAILED);
     return;
   }
 
-  if (!shared_contribution) {
+  if (!contribution) {
```

Comment:
I noticed this at a few places: such a check was not present previously (would have been `if (!*shared_contribution) { /* ... */ }`). But then I don't think we rely on the side effects anywhere (or if there's any side effects at all).

Reply:
It changes the condition to check at one less fewer levels of indirection, which seems fine to me.

Reply:
👍🏻